libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2312,7 +2312,11 @@ bool TypeChecker::visit(FunctionCallOptions const& _functionCallOptions)
 			else if (!expressionFunctionType->isPayable())
 				m_errorReporter.typeError(
 					_functionCallOptions.location(),
-					"Cannot set option \"value\" on a non-payable function type."
+					kind == FunctionType::Kind::Creation ?
+						"Constructor for contract " +
+						expressionFunctionType->returnParameterTypes().front()->toString() +
+						" must be payable for member \"value\" to be available." :
+						"Member \"value\" is only available for payable functions."
```

Comment:
I think we may have miscommunicated about this a bit - I rather meant something like this:
```suggestion
					kind == FunctionType::Kind::Creation ?
						"Cannot set option \"value\", since the constructor of " +
						expressionFunctionType->returnParameterTypes().front()->toString() +
						" is not payable." :
						"Cannot set option \"value\" on a non-payable function type."
```

By the way: I trust that Chris has already shown you how to use ``isoltest`` to automatically update these, right :-)?

Reply:
Of course, it is the best time saving feature :-D

Ok, let's do like this, but what to do with the same error in MemberAcess part? To be similar like here?

Reply:
The error in ``MemberAccess`` is fine - in that case it's in fact a "member" that is not available.
Earlier I just meant that the constructor case should - similarly to the one in ``MemberAccess```- not just talk about a "function", but about the constructor. So I'd say it's fine now!

Reply:
Okay, updating it now :)
Let's see if I missed some tests after CI is done.